File: Starters 59/Confusing_Concatenations.cpp
// -----headers-----

#include<bits/stdc++.h>
#include<iostream>
// #include<vector>
// #include<math.h>

// -----templates-----

#define fast_io ios_base::sync_with_stdio(false);cin.tie(NULL);cout.tie(NULL);
#define ll long long
#define ld long double
#define MAX INT_MAX
#define MIN INT_MIN
#define LMIN -10e18
#define LMAX 10e18

#define pb(x) push_back(x)
#define eb(x) emplace_back(x)

using namespace std;

void init_code()
{
	fast_io;
	#ifndef ONLINE_JUDGE
	freopen("input.txt", "r", stdin);
	freopen("output.txt", "w", stdout);
	#endif
}
void solve()
{
	ll n; 
	cin >> n; 
	ll arr[n]; 

	ll maxVal = MIN; 
	ll index = -1; 

	for(ll i=0; i<n; ++i)
	{
		cin >> arr[i]; 

		if(arr[i] > maxVal)
		{
			maxVal = arr[i]; 
			index = i;
		}
	}

	if(index == 0)
		cout << -1 << endl; 

	else
	{
		cout << index << endl;
		for(ll i=0; i<index; ++i)
		{
			cout << arr[i] << " ";
		}

		cout << endl; 

		cout << (n-index) << endl;

		for(ll i=index; i<n; ++i)
			cout << arr[i] << " ";

		cout << endl; 
	}
}
int main()
{
	// init_code();
	ll t; 
	cin >> t;
	while(t--)
	{
		solve();
	}

	// for(ll i=1; i <= t; i++)
	// 	cout << solve(i) << endl;
	return 0;
}

File: Starters 59/Confusing_Concatenations.java


            // ******************* TLE *******************

import java.util.*;
class Confusing_Concatenations
{
	static Scanner ob = new Scanner(System.in);
	private int findMaxIndex(int arr[], int n)
	{
		int maxVal = arr[0];
		int index = 0;

		for(int i=0; i<n; i++)
		{
			if(arr[i] > maxVal)
			{
				maxVal = arr[i]; 
				index = i;
			}
		}
		return index; 
	}
	public void solve()
	{
		int n = ob.nextInt(); 
		int arr[] = new int[n]; 

		for(int i=0; i<n; i++)
		{
			arr[i] = ob.nextInt();
		}

		int a = findMaxIndex(arr, n);
		
		if(a == 0)
			System.out.println(-1); 

		else
		{
			System.out.println(a);
			for(int i=0; i<a; i++)
				System.out.print(arr[i] + " ");

			System.out.println(); 

			System.out.println(n-a);
			for(int i=a; i<n; i++)
				System.out.print(arr[i] + " ");

			System.out.println();
		}
	}
	
	public static void main(String[] args)
	{
		// ----------template--------------

		// if (System.getProperty("ONLINE_JUDGE") == null) {
			
		// 	try 
		// 	{
		// 		System.setOut(new PrintStream(new FileOutputStream("output.txt")));
		// 		ob = new Scanner(new File("input.txt"));
		// 	}
		// 	catch (Exception e){}
		// }

		// ----------------------------------

		int t = ob.nextInt();
		while(t-- > 0)
		{
			new Confusing_Concatenations().solve();
			// System.out.println(new Solution().solve());
		}

		// System.out.println(inp + " World!");
	}
}